### DISCUSSION TITLE
Need unzip option

### DISCUSSION THREAD

**Discussion by mohitjoshi155** [filebrowser/filebrowser#1585] [Ideas]
can unzip option be added plz so we can unzip files on filebrowser

**Comment by ChilliGeologist** [comment]
I second this, support .rar as well would be amazing but at least .zip would be great

**Comment by hoangnamitc** [comment]
+1 vote for this feature!

Thanks

**Comment by quocthinhvo** [comment]
You can enable command for user and add "zip unzip" command. Open the terminal in FileBrowser and type: unzip filename.zip.
Note: please sure u have install zip package or zip program. 
Good luck!

**Comment by hernandito** [comment]
+1 Vote. Also include .rar, tar.

 I will test the command line…. Will this work in a Docker on a Linux system? But it would be great to be able to do this on the GUI… sometimes those zip and rar files can have long, long names…

**Comment by gingerbeardman** [comment]
Use of `unzip` is already possible, but it is disabled by default:

## Adding zip command to admin user
1. go to Settings > User Management > admin > edit (pencil icon) > (scroll down) Commands > type `unzip` > click Update
2. use filebrowser to find your zip file
3. open the console (`< >` toolbar button)
4. type `unzip file`
5. watch unzip progress: you're done!

### Optional, to set commands for new users
1. go to Settings > Global Settings > (scroll down) User default settings > Commands > type `unzip` > click Update
2. create new user
3. commands specified in step 1 are already assigned and available

Note: you are limited to commands present in the host system. In my case using the `hurlenko/filebrowser` docker container the host system is [BusyBox, which has this list of commands](https://boxmatrix.info/wiki/BusyBox-Commands)

...it would be nice to have an unzip button but that requires changes to `filebrowser` itself.

...but it needs to take into consideration that the server you are running on may not have `unzip`.

**Reply by zagi988** [reply]
+1

**Reply by medram** [reply]
unzipping archives should be added in the UI.

**Reply by gingerbeardman** [reply]
it's more complicated than "just" adding to the UI, because the system/server might not have unzip

**Reply by medram** [reply]
Just install the "unzip" requirement/dependency before building the docker image, and it should be easy to implement in the UI.

**Reply by gingerbeardman** [reply]
See the other comments, it has already been done. PR #2233 is awaiting merge.

**Reply by rj10rd** [reply]
> Use of `unzip` is already possible, but it is disabled by default:
> 
> ## Adding zip command to admin user
> 1. go to Settings > User Management > admin > edit (pencil icon) > (scroll down) Commands > type `unzip` > click Update
> 2. use filebrowser to find your zip file
> 3. open the console (`< >` toolbar button)
> 4. type `unzip file`
> 5. watch unzip progress: you're done!
> 
> ### Optional, to set commands for new users
> 1. go to Settings > Global Settings > (scroll down) User default settings > Commands > type `unzip` > click Update
> 2. create new user
> 3. commands specified in step 1 are already assigned and available
> 
> Note: you are limited to commands present in the host system. In my case using the `hurlenko/filebrowser` docker container the host system is [BusyBox, which has this list of commands](https://boxmatrix.info/wiki/BusyBox-Commands)
> 
> ...it would be nice to have an unzip button but that requires changes to `filebrowser` itself.
> 
> ...but it needs to take into consideration that the server you are running on may not have `unzip`.

Unzip command not working in docker

**Comment by gingerbeardman** [comment]
Just checking... and this has been done by @t924417424 - thanks! See PR https://github.com/filebrowser/filebrowser/pull/2233 dated 2022-11-22

It is current sitting waiting for approval/review.

**Reply by iceunlim** [reply]
seems it has been closed,what a pity :(

**Reply by gingerbeardman** [reply]
Yes, this software is largely in maintenance-only mode. Nobody is interested in merging PRs or adding new features. The automatic bot is very destructive closing issues and PRs.

**Reply by dreamingbird88** [reply]
Instead of install the standard version, is there any instruction on how to install this version with 'unzip' support? Thanks.

**Comment by khairul169** [comment]
If anyone still need this feature, i've built custom docker image with PR #2233 by @t924417424
`khairul169/filebrowser:latest`

Commit: c6139c4798c277241d4892969e1507f1da8b0d8e

**Reply by gingerbeardman** [reply]
Awesome! Do you plan to keep it up-to-date with future changes?

**Reply by MrBooro** [reply]
Friend, thank you very much!!!! You saved my life.

**Comment by bloodykheeng** [comment]
@gingerbeardman thanks though u forgot to inform that one has to first install unzip package in linux

# How to Unzip Files in FileBrowser

first check if unzip is installed on ur system

```bash
unzip -v
```

if not installed then istall it using this comand

```
sudo apt install unzip -y
```

then after come back to file browser

## Step 1: Enable the `unzip` Command for Admin Users

1. Navigate to **Settings > User Management**.
2. Select the **admin** user and click the **edit (pencil icon)**.
3. Scroll down to **Commands**.
4. Type `unzip` in the Commands field.
5. Click **Update** to save changes.

## Step 2: Unzip Files Using FileBrowser

1. Use FileBrowser to locate your **.zip** file.
2. Open the **console** (click the `< >` toolbar button).
3. Type the following command:
   ```sh
   unzip filename.zip
   ```
   _(Replace `filename.zip` with the actual file name)_
4. Watch the unzip progress – you're done!

**Reply by gingerbeardman** [reply]
Great! I didn't need to install unzip. Glad you got it working.
